packages/semantic-model/src/api-json.d.ts

```diff
@@ -66,12 +66,14 @@ export type Exceptions = {
  * Namespaces can have the common symbol properties and extend another symbol
  */
 export type NamespaceSymbol = SymbolBase & {
-  kind?: "namespace";
+  kind?: "namespace" | "member";
```

Comment:
This is strange, what is this "member"? is this a mistake in the JSDocs?
Maybe a pre-processor is needed for the API.json files so the schema would not contain  invalid data?

Reply:
This is particularly suspicious as we did not need it for version 1.74, so it seems like a bug that was fixed.

Reply:
I agree that it's probably a bug in the jsdoc. The question is if it's necessary to pre-fix it or if it's better to handle it in the schema and conversion code like now. There are other likely bugs in the jsdoc that we probably aren't going to pre-fix (like extra fields on some types that I also added to the schema).

Reply:
What are the capabilities of schema merging or union types and schema --> d.ts generation?

I'd like to have some distinction between valid things and "ignored issues", e.g:

```typescript
interface namespace {
  kind?: "namespace" | InvalidNamespaceKind
}

export type InvalidNamespaceKind = "member"
```

Basically we need to distinguish between what is valid and what is not, otherwise our Schema will no longer be very useful in the long term.

Reply:
Everything under `definitions` gets a named type as far as I can tell. It might be possible to add an enum there with a single value and reference it from the namespace definition.
The question is what you define as "not valid" in the schema. Because while this looks like an unintended bug, it does actually exist in the api.json files (at least in some versions), and there is no "real" or "valid" version of the schema anywhere. The original schema had json syntax errors and even after fixing them I had to change it so it will capture what's really in the api.json files (not just the member thing, there were many other changes). Who's to say that not all of these fixes are errors in the jsdoc?

Reply:
The problem is:
1. Overtime we will add more and more "exceptions" due to human errors and more UI5 versions supported.
2. We won't know what parts of the schema are errors and what are not.

Handling this issue can be deferred to another time, however we should not forget this technical debt. I think we may need to do a push to move this schema to the UI5 build
to fail fast on human errors...

@codeworrior FYI

Reply:
I started an email discussion about this. lets defer it to the future.